bootstrap.sh

```
@@ -104,7 +110,63 @@ function runCmdAndLog() {
+}
+
+# Log a step message with step counter and current date/time
+# Usage: logStep <message>
+#   message: The message to log
+#   Increments the step counter and logs the message with the step number
+#   Example: logStep "Installing tools..."
+#     Logs: 2023-10-01 12:34:56 - [ 1/13] Installing tools...
+function logStep() {
+    CURRENT_STEP=$((CURRENT_STEP + 1))
+    printf -v STEP_NUM "%2d" "${CURRENT_STEP}"
+    log "[${STEP_NUM}/${TOTAL_STEPS}] ${@}"
```

Shellcheck (0.11.0)

[error] Argument mixes string and array. Use * or separate argument.

(SC2145)